### PR TITLE
feat: update backend to support MDS

### PR DESCRIPTION
**Pull request by raintygao**
### Description
1. Introduce DSM and DS as dependency
2. update router and controller to support MDS

There will be another PR about integrating in frontend and depend on https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6827

### Issue resolved
https://github.com/opensearch-project/dashboards-assistant/issues/192

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test.
- [ ] New functionality has user manual doc added.
- [ ] Commits are signed per the DCO using --signoff.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/hooks/__tests__/use_conversations.test.ts [hunk 1]
```diff
@@ -7,17 +7,36 @@ import { useDeleteConversation, usePatchConversation } from '../use_conversation
 import { renderHook, act } from '@testing-library/react-hooks';
 import { useCore } from '../../contexts/core_context';
 import { HttpHandler } from '../../../../../src/core/public';
-
+import { DataSourceServiceMock } from '../../services/data_source_service.mock';
+import { httpServiceMock } from '../../../../../src/core/public/mocks';
 jest.mock('../../contexts/core_context');
-const useCoreMocked = useCore as jest.MockedFunction<typeof useCore>;
+
+const setup = () => {
```
Comment (wanglam) [inline public/hooks/__tests__/use_conversations.test.ts:14]: There is already one mock file for core_context in `public/contexts/__mocks__/core_context.tsx`, can we add dataSource mock to this mock? So I think we don't need this setup function. We can use the `useCoreMocked` directly.

File: public/services/data_source_service.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BehaviorSubject } from 'rxjs';
```
Comment (wanglam) [inline public/services/data_source_service.ts:6]: Seems these imports are not been used.

File: server/routes/chat_routes.ts [hunk 3]
```diff
@@ -123,16 +146,20 @@ const feedbackRoute = {
     body: schema.object({
       satisfaction: schema.boolean(),
     }),
+    query: schema.object({
+      dataSourceId: schema.maybe(schema.string()),
```
Comment (wanglam) [inline server/routes/chat_routes.ts:150]: Shall we add some UT for dataSourceId query passed?

Comment (raintygao) [reply server/routes/chat_routes.ts:150]: We could add UT for data source specifically, for now this PR is depended by other PRs, right?

File: public/components/feedback_modal.tsx
[the remark this answers is not in the document]
Comment (raintygao) [reply public/components/feedback_modal.tsx:1]: Yes. This file isn't used but contain API request, we should make code clear.

File: public/chat_flyout.tsx [hunk 4]
```diff
@@ -22,7 +25,15 @@ interface ChatFlyoutProps {
 export const ChatFlyout = (props: ChatFlyoutProps) => {
   const chatContext = useChatContext();
   const chatHistoryPageLoadedRef = useRef(false);
-
+  const core = useCore();
+  // TODO: use DataSourceService to replace
+  const selectedDS = useObservable(
+    core.services.setupDeps?.dataSourceManagement?.dataSourceSelection?.getSelection$() ??
+      new BehaviorSubject(new Map())
```
Comment (wanglam) [inline public/chat_flyout.tsx:32]: This new BehaviorSubject cause many re-render in my local env, shall we fix this?

File: server/routes/chat_routes.ts (hunk 3 above)
Comment (wanglam) [reply server/routes/chat_routes.ts:150]: Do you mean will be added in the following PRs?

Comment (raintygao) [reply server/routes/chat_routes.ts:150]: I remember that you wanted this PR being merged ASAP, if not, we could try to add tests in this PR to make this fully.

Comment (wanglam) [reply server/routes/chat_routes.ts:150]: I think you can added them first. At least we need to verify the route handler called getOpenSearchClientTransport with passed data source id.

**Review by wanglam — APPROVED** [review]

File: public/chat_flyout.tsx [hunk 5]
```diff
@@ -22,7 +25,17 @@ interface ChatFlyoutProps {
 export const ChatFlyout = (props: ChatFlyoutProps) => {
   const chatContext = useChatContext();
   const chatHistoryPageLoadedRef = useRef(false);
+  const core = useCore();
+  const defaultDataSourceSelection$ = new BehaviorSubject(new Map());
```
Comment (SuZhou-Joe) [inline public/chat_flyout.tsx:29]: Should we move this line out of the ChatFlyout?

File: public/chat_flyout.tsx [hunk 6]
```diff
@@ -87,6 +100,10 @@ export const ChatFlyout = (props: ChatFlyoutProps) => {
       <>
         <div className={cs('llm-chat-flyout-header')}>
           <ChatWindowHeader />
+          <div>
+            Data Source:
+            {currentDS?.[0]?.label}
+          </div>
```
Comment (SuZhou-Joe) [inline public/chat_flyout.tsx:106]: Can we add a screenshot on how will this UI change will be displayed?

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**